File: Braintree/src/test/java/com/braintreepayments/api/models/VenmoConfigurationUnitTest.java
package com.braintreepayments.api.models;

import android.text.TextUtils;

import com.braintreepayments.api.test.VenmoInstalledContextFactory;

import org.json.JSONException;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.RobolectricTestRunner;

import static com.braintreepayments.api.internal.SignatureVerificationUnitTestUtils.disableSignatureVerification;
import static com.braintreepayments.testutils.FixturesHelper.stringFromFixture;
import static junit.framework.Assert.assertEquals;
import static junit.framework.Assert.assertFalse;
import static junit.framework.Assert.assertTrue;

@RunWith(RobolectricTestRunner.class)
public class VenmoConfigurationUnitTest {

    private Configuration mConfiguration;

    @Before
    public void setup() throws JSONException {
        mConfiguration = Configuration.fromJson(stringFromFixture("configuration/with_pay_with_venmo.json"));
    }

    @Test
    public void isVenmoWhitelisted_returnsTrueForWhitelist() {
        assertTrue(mConfiguration.getPayWithVenmo().isVenmoWhitelisted(VenmoInstalledContextFactory
                .venmoInstalledContext(true).getContentResolver()));
    }

    @Test
    public void fromJson_parsesPayWithVenmoConfiguration() throws JSONException {
        assertEquals("access-token", mConfiguration.getPayWithVenmo().getAccessToken());
        assertEquals("environment", mConfiguration.getPayWithVenmo().getEnvironment());
        assertEquals("merchant-id", mConfiguration.getPayWithVenmo().getMerchantId());
    }

    @Test
    public void fromJson_parsesEmptyVenmoConfigurationWhenConfigurationDoesntHavePayWithVenmo() throws JSONException {
        Configuration configuration = Configuration.fromJson(stringFromFixture("configuration/configuration.json"));

        assertEquals("", configuration.getPayWithVenmo().getAccessToken());
        assertTrue(TextUtils.isEmpty(configuration.getPayWithVenmo().getAccessToken()));
    }

    @Test
    public void isEnabled_returnsTrueWhenEnabled() throws JSONException {
        assertFalse(TextUtils.isEmpty(mConfiguration.getPayWithVenmo().getAccessToken()));
    }

    @Test
    public void isEnabled_returnsTrueWhenAppIsInstalled() throws JSONException {
        disableSignatureVerification();
        assertTrue(mConfiguration.getPayWithVenmo().isEnabled(VenmoInstalledContextFactory.venmoInstalledContext(true)));
    }

    @Test
    public void isEnabled_returnsFalseForNotInstalled() {
        assertFalse(mConfiguration.getPayWithVenmo().isEnabled(VenmoInstalledContextFactory.venmoInstalledContext(false)));
    }
}

File: BraintreeCore/src/test/java/com/braintreepayments/api/core/VenmoConfigurationUnitTest.kt
package com.braintreepayments.api.core

import org.robolectric.RobolectricTestRunner
import org.json.JSONException
import org.json.JSONObject
import org.junit.Assert.*
import org.junit.Test
import org.junit.runner.RunWith

@RunWith(RobolectricTestRunner::class)
class VenmoConfigurationUnitTest {
    @Test
    @Throws(JSONException::class)
    fun `when json contains all fields, VenmoConfiguration parses all values and isAccessTokenValid is true`() {
        val input = JSONObject()
            .put("accessToken", "sample-access-token")
            .put("environment", "sample-environment")
            .put("merchantId", "sample-merchant-id")
        val sut = VenmoConfiguration(input)
        assertEquals("sample-access-token", sut.accessToken)
        assertEquals("sample-environment", sut.environment)
        assertEquals("sample-merchant-id", sut.merchantId)
        assertTrue(sut.isAccessTokenValid)
    }

    @Test
    fun `when input json is null, VenmoConfiguration has empty fields and isAccessTokenValid is false`() {
        val sut = VenmoConfiguration(null)
        assertEquals("", sut.accessToken)
        assertEquals("", sut.environment)
        assertEquals("", sut.merchantId)
        assertFalse(sut.isAccessTokenValid)
    }

    @Test
    fun `when input json is empty, VenmoConfiguration has empty fields and isAccessTokenValid is false`() {
        val sut = VenmoConfiguration(JSONObject())
        assertEquals("", sut.accessToken)
        assertEquals("", sut.environment)
        assertEquals("", sut.merchantId)
        assertFalse(sut.isAccessTokenValid)
    }
}
